common/meta/types/src/change.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_exception::ErrorCode;
+use serde::Deserialize;
+use serde::Serialize;
+
+use crate::SeqV;
+
+pub enum AddResult<T> {
+    Ok(SeqV<T>),
+    Exists(SeqV<T>),
+}
+
+/// `Change` describes a state change, including the states before and after a change.
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, derive_more::From)]
+pub struct Change<T>
```

Comment:
I think the name `Exchange` is more appropriate here.

Reply:
Hm... not quite sure...